crates/lib/core/tests/crypto/falcon.rs

```diff
@@ -400,16 +518,29 @@ fn generate_data_probabilistic_product_test(
     // get the challenge point and push it to the advice stack
     let digest_polynomials = Rpo256::hash_elements(&polynomials);
     let challenge = (digest_polynomials[0], digest_polynomials[1]);
-    let mut advice_stack = vec![challenge.0.as_int(), challenge.1.as_int()];
+    let mut advice_stack = vec![challenge.0.as_canonical_u64(), challenge.1.as_canonical_u64()];
 
     // push the polynomials to the advice stack
-    let polynomials: Vec<u64> = polynomials.iter().map(|&e| e.into()).collect();
+    let polynomials: Vec<u64> = polynomials.iter().map(|&e| e.as_canonical_u64()).collect();
+
+    // Debug: check first few values
+    eprintln!(
+        "Challenge values: [{}, {}]",
+        challenge.0.as_canonical_u64(),
+        challenge.1.as_canonical_u64()
+    );
+    eprintln!("First 10 polynomial values on advice stack:");
+    for (i, &val) in polynomials.iter().take(10).enumerate() {
+        eprintln!("  [{}] = {} (< M={}: {})", i, val, M, val < M);
+    }
+    eprintln!("Total polynomial count: {}", polynomials.len());
```

Comment:
Do we still need these print statements here?

Reply:
Removed in https://github.com/0xMiden/miden-vm/pull/2472/commits/f2dac9e449af1dee223f18abffb4a9758b8441ce